packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx

```diff
@@ -224,12 +219,64 @@ const TradeParamsMobile = observer(
         h_duration,
         d_duration,
     }: TTradeParamsMobile) => {
-        const { basis_list, basis, expiry_epoch, is_turbos, is_vanilla } = useTraderStore();
+        const {
+            basis_list,
+            basis,
+            expiry_epoch,
+            is_turbos,
+            is_vanilla,
+            duration_unit: store_duration_unit,
+        } = useTraderStore();
+        React.useEffect(() => {
+            const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+            const default_duration_tab_idx = duration_units_list.findIndex(d => d.value === store_duration_unit);
+            const defaultDuration = getDefaultDuration(duration_unit);
+
+            // reset to default value and timeframe when previously chosen timeframe doesnt exist in new trade type
+            if (toggled_duration_tab_idx === -1) {
+                const defaultDuration = getDefaultDuration(store_duration_unit);
+                setSelectedDuration(store_duration_unit, defaultDuration);
+            }
+            setSelectedDuration(duration_unit, has_duration_error ? defaultDuration : duration_value);
+            setDurationTabIdx(toggled_duration_tab_idx === -1 ? default_duration_tab_idx : toggled_duration_tab_idx);
+
+            return () => {
+                setSelectedDuration(duration_unit, has_duration_error ? defaultDuration : duration_value);
+            };
+        }, [duration_unit]);
+
+        React.useEffect(() => {
+            const defaultDuration = getDefaultDuration(duration_unit);
+            const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+
+            if (toggled_duration_tab_idx === -1) {
+                const defaultDuration = getDefaultDuration(store_duration_unit);
+                setSelectedDuration(store_duration_unit, defaultDuration);
+            }
+            return () => {
+                setSelectedDuration(duration_unit, has_duration_error ? defaultDuration : duration_value);
+            };
+        }, [duration_value]);
+
         const getDurationText = () => {
-            const duration = duration_units_list.find(d => d.value === duration_unit);
-            return `${duration_value} ${
-                duration && (duration_value > 1 ? localize(duration.text) : localize(duration.text.slice(0, -1)))
-            }`;
+            const default_duration_obj = duration_units_list.find(d => d.value === store_duration_unit);
+            const toggled_duration_obj = duration_units_list.find(d => d.value === duration_unit);
+            const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+            const defaultDuration = getDefaultDuration(store_duration_unit);
+            const default_timeframe =
```

Comment:
`defaultDuration` but `default_timeframe`... :) it'd be better to stick to camelCase then

Reply:
changed them to snake_case since they are values